### PR TITLE
Add Params and resourceManagerTags for instance and disks

### PR DESCRIPTION
**Pull request by zqc764487**
Add Params and resourceManagerTags for instance and disks with tests.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: Added general field `params` to `google_compute_instance`

```

### CODE REVIEW

**Comment by melinath** [comment]
Running VCR recording in TeamCity to avoid timeouts: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloudBeta_ProviderGoogleCloudBetaMmUpstreamVcr/357059

**Comment by shuyama1** [comment]
Check if cassettes got pushed correctly. Should have only a small number of tests failed in replaying.

/gcbrun

File: mmv1/products/compute/api.yaml [hunk 1]
```diff
@@ -6561,6 +6561,14 @@ objects:
                     disks.source is required.  To create a disk with one of
                     the public operating system images, specify the image
                     by its family name.
+                - !ruby/object:Api::Type::KeyValuePairs
```
Comment (rileykarson) [inline mmv1/products/compute/api.yaml:6564]: You can revert the changes to this file- they don't do anything, as MMv1 doesn't generate this resource (although it's used to handle the copied files)

File: mmv1/products/compute/terraform.yaml [hunk 2]
```diff
@@ -1382,6 +1382,11 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         vars:
           instance_name: "my-instance"
         primary_resource_name: "fmt.Sprintf(\"tf-test-my-instance%s\", context[\"random_suffix\"])"
+      - !ruby/object:Provider::Terraform::Examples
```
Comment (rileykarson) [inline mmv1/products/compute/terraform.yaml:1385]: Same as above- this doesn't do anything and can be removed.

File: mmv1/templates/terraform/examples/instance_with_atm_tags.tf.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+resource "google_compute_instance" "instance_with_atm_tags" {
```
Comment (rileykarson) [inline mmv1/templates/terraform/examples/instance_with_atm_tags.tf.erb:1]: Same as above- this doesn't do anything and can be removed.

File: mmv1/third_party/terraform/resources/resource_compute_instance.go.erb [hunk 4]
```diff
@@ -1004,6 +1034,11 @@ func expandComputeInstance(project string, d *schema.ResourceData, config *Confi
 		return nil, fmt.Errorf("Error creating scheduling: %s", err)
 	}
 
+        params, err := expandParams(d)
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_instance.go.erb:1037]: nit: Do you mind renaming associated expand / flatten funcs to namespace on the resource, similar to https://github.com/hashicorp/terraform-provider-google/blob/53295dc35039ca8cd66da18cc758bf7487bd8c03/google/resource_compute_address.go#L195-L200?

This file predates that convention, unfortunately, so there are counterexamples present.

File: mmv1/third_party/terraform/resources/resource_compute_instance.go.erb [hunk 5]
```diff
@@ -216,6 +217,15 @@ func resourceComputeInstance() *schema.Resource {
 										ForceNew:     true,
 										Description:  `A set of key/value label pairs assigned to the disk.`,
 									},
+
+									"resource_manager_tags": {
+										Type:         schema.TypeMap,
+										Optional:     true,
+										AtLeastOneOf: initializeParamsKeys,
+										Computed:     true,
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_instance.go.erb:225]: Why is this set to `Computed: true`? (& same for the new fields below)

File: mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb [hunk 6]
```diff
@@ -226,6 +226,31 @@ func TestAccComputeInstance_basic5(t *testing.T) {
...
+	var instanceName = fmt.Sprintf("tf-test-%s", randString(t, 10))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeInstance_basic_resourceManagerTags(instanceName),
+				Check: resource.ComposeTestCheckFunc(
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb:242]: You can remove all `Check` funcs, but we'll want to add an import step- see `TestAccComputeInstance_basic1 `, the `computeInstanceImportStep` invocation. As above, this is an old file and comes from a time when we still used `Check`s. Import largely superseded it.

File: mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb [hunk 7]
```diff
@@ -3302,6 +3349,37 @@ resource "google_compute_instance" "foobar" {
 `, instance)
 }
 
+func testAccComputeInstance_basic_resourceManagerTags(instance string) string {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb:3352]: This doesn't appear to have set the new fields?

File: mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb [hunk 8]
```diff
@@ -226,6 +226,31 @@ func TestAccComputeInstance_basic5(t *testing.T) {
 	})
 }
 
+func TestAccComputeInstance_basic_resourceManagerTags(t *testing.T) {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb:229]:
```suggestion
func TestAccComputeInstance_resourceManagerTags(t *testing.T) {
```

(we'll want to similarly rename the config function)

File: mmv1/third_party/terraform/resources/resource_compute_instance.go.erb (hunk 5 above)
Comment (zqc764487) [reply mmv1/third_party/terraform/resources/resource_compute_instance.go.erb]: Yes, I guess this is unnecessary

File: mmv1/third_party/terraform/resources/resource_compute_instance.go.erb (hunk 4 above)
Comment (zqc764487) [reply mmv1/third_party/terraform/resources/resource_compute_instance.go.erb:1035]: It makes sense to me but in this folder I guess it would be better to follow the same name convention as we can see "expandBootDisk" etc. here?

File: mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb (hunk 7 above)
Comment (zqc764487) [reply mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb:3352]: Yes, I tested in my local provider and made sure the new fields worked fine, the reason why I did not add these new fields is because these fields are still in test stage and only allowlisted some specific projects. I will add these new fields once these fields are exposed to all projects

File: mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb (hunk 6 above)
Comment (zqc764487) [reply mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb:242]: ACK. Can we keep them as it is for now? I can inform GCE team, the owner of this file, to refactor it later.

Comment (rileykarson) [reply mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb:242]: The provider codebase is owned by my team. Our guidance for new tests is to use import steps, see https://googlecloudplatform.github.io/magic-modules/docs/how-to/add-handwritten-test/#simple-tests for an example.

File: mmv1/third_party/terraform/resources/resource_compute_instance.go.erb [hunk 9]
```diff
@@ -540,6 +549,25 @@ func resourceComputeInstance() *schema.Resource {
 				},
 			},
 
+	                "params": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_instance.go.erb:557]:
```suggestion
```

I believe this case of `Computed` is unnecessary as well?

File: mmv1/third_party/terraform/resources/resource_compute_instance.go.erb [hunk 10]
```diff
@@ -540,6 +549,25 @@ func resourceComputeInstance() *schema.Resource {
...
+				Description: `Stores additional params passed with the request, but not persisted as part of resource payload.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"resource_manager_tags": {
+							Type:         schema.TypeMap,
+							Optional:     true,
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_instance.go.erb:563]: Should we consider making the field required? Would a user every want to specify `params {}`?

File: mmv1/third_party/terraform/resources/resource_compute_instance.go.erb (hunk 4 above)
Comment (rileykarson) [reply mmv1/third_party/terraform/resources/resource_compute_instance.go.erb:1035]: Ah- I guess the whole file's consistent in not doing so. SGTM.

File: mmv1/third_party/terraform/resources/resource_compute_instance.go.erb (hunk 10 above)
Comment (zqc764487) [reply mmv1/third_party/terraform/resources/resource_compute_instance.go.erb:562]: I think we should leave it optional, because params will have some more fields to be exposed as optional.

File: mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb (hunk 6 above)
Comment (rileykarson) [reply mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb:242]: You can remove the existence check as well- that's also done by the import step.

File: mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb (hunk 8 above)
Comment (rileykarson) [reply mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb:229]: This still needs to be changed!

File: mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb [hunk 11]
```diff
@@ -2876,6 +2899,28 @@ func testAccCheckComputeInstanceLabel(instance *compute.Instance, key string, va
 	}
 }
 
+func testAccCheckComputeInstanceParamResourceManagerTag(instance *compute.Instance, key string, value string) resource.TestCheckFunc {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb:2902]: Now that this function is unused, we can remove it.

File: mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb (hunk 7 above)
Comment (rileykarson) [reply mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb:3350]: Sorry, I missed this before because it was resolved. We can't move forward with this PR until the fields are made public, I'll follow up internally.

**Review by rileykarson — COMMENTED** [review]
Noting in the mainline comment thread, in addition to review comments, this change is blocked on the fields being made available publicly without an allowlist.

**Comment by rileykarson** [comment]
This was merged as https://github.com/GoogleCloudPlatform/magic-modules/pull/8107
